src/mlpack/methods/kde/kde_main.cpp

```diff
@@ -48,7 +48,21 @@ PROGRAM_INFO("Kernel Density Estimation",
     "type of tree to use for the dual-tree algorithm with " +
     PRINT_PARAM_STRING("tree") + ". It is also possible to select whether to "
     "use dual-tree algorithm or single-tree algorithm using the " +
-    PRINT_PARAM_STRING("algorithm") + " option."
+    PRINT_PARAM_STRING("algorithm") + " option. This implementation makes "
```

Comment:
Given how long the section you added is, it may make sense to split the Monte Carlo discussion into a separate paragraph.